src/objects/zcl_abapgit_object_iaxu.clas.abap

```diff
@@ -0,0 +1,309 @@
+      w3_api_load
+        IMPORTING is_name    TYPE iacikeyt
+        EXPORTING eo_xml_api TYPE REF TO cl_w3_api_xml3
+                  es_attr    TYPE w3tempattr
+        RAISING   zcx_abapgit_exception,
+      w3_api_set_changeable
+        IMPORTING io_xml_api    TYPE REF TO cl_w3_api_xml3
+                  iv_changeable TYPE abap_bool
+        RAISING   zcx_abapgit_exception,
+      w3_api_delete
+        IMPORTING io_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception,
+      w3_api_save
+        IMPORTING io_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception,
+      w3_api_create_new
+        IMPORTING is_attr    TYPE w3tempattr
+        EXPORTING eo_xml_api TYPE REF TO cl_w3_api_xml3
+        RAISING   zcx_abapgit_exception.
+
+ENDCLASS.
+
+
+
+CLASS zcl_abapgit_object_iaxu IMPLEMENTATION.
+
+
+  METHOD read.
+
+    DATA: ls_name TYPE iacikeyt.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name = ls_name
+                 IMPORTING es_attr = es_attr ).
+
+    CLEAR: es_attr-chname,
+           es_attr-tdate,
+           es_attr-ttime,
+           es_attr-devclass.
+
+  ENDMETHOD.
+
+
+  METHOD save.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3.
+
+
+    w3_api_create_new( EXPORTING is_attr    = is_attr
+                       IMPORTING eo_xml_api = lo_xml_api ).
+
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~changed_by.
+    rv_user = c_user_unknown. " todo
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~delete.
+
+    DATA: lo_xml_api TYPE REF TO cl_w3_api_xml3,
+          ls_name    TYPE iacikeyt,
+          lx_root    TYPE REF TO zcx_abapgit_exception.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name    = ls_name
+                 IMPORTING eo_xml_api = lo_xml_api ).
+
+    IF lo_xml_api IS NOT BOUND.
+      zcx_abapgit_exception=>raise( |Error from w3_api_xml3~load subrc={ sy-subrc }| ).
+    ENDIF.
+
+    w3_api_set_changeable( io_xml_api    = lo_xml_api
+                           iv_changeable = abap_true ).
+
+    w3_api_delete( io_xml_api = lo_xml_api ).
+
+    w3_api_save( io_xml_api = lo_xml_api ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~deserialize.
+
+    DATA: ls_attr TYPE w3tempattr.
+
+
+    io_xml->read( EXPORTING iv_name = 'ATTR'
+                  CHANGING  cg_data = ls_attr ).
+
+    ls_attr-devclass = iv_package.
+
+    IF zif_abapgit_object~exists( ) = abap_true.
+      zif_abapgit_object~delete( ).
+    ENDIF.
+
+    save( is_attr = ls_attr ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~exists.
+
+    DATA: ls_name    TYPE iacikeyt,
+          lo_xml_api TYPE REF TO cl_w3_api_xml3.
+
+
+    ls_name = ms_item-obj_name.
+
+    w3_api_load( EXPORTING is_name = ls_name
+                 IMPORTING eo_xml_api = lo_xml_api ).
+
+    IF lo_xml_api IS BOUND.
+      rv_bool = abap_true.
+    ELSE.
+      rv_bool = abap_false.
+    ENDIF.
```

Comment:
I can't do that in the exists-method because some exceptions (sy-subrc > 1) have another meaning than "object is missing" (sy-subrc = 1). If I catch ZCX_ABAPGIT_EXCEPTION I don't know what happened in the underlying method which throws the exception. I could analyse the exception text which holds sy-subrc but that seems a bit too much. Am I missing something?

Reply:
I've implemented an alternative approach. Also thanks for your continuous reviews and suggestions! :)

Reply:
Imho it would be sufficient to interpret every catched exception as `rv_bool = abap_false` and not differentiate, but I agree that's not the cleanest solution.

What about using `CL_W3_API_XML3=>S_CHECK_EXIST` method instead of ` W3_API_LOAD`? Then we can get rid of `EV_SUBRC`  and simplify the exception handling in `EXISTS` further.

Reply:
That simplifies a lot..